x/wasm/types/errors_test.go

```diff
@@ -94,3 +94,19 @@ func TestWasmVMFlavouredError(t *testing.T) {
 		t.Run(name, spec.exec)
 	}
 }
+
+func TestDeterministicError(t *testing.T) {
+	inner := ErrInstantiateFailed
+	err := MarkErrorDeterministic(inner)
+
+	// behaves like a wrapper around inner error
+	assert.Equal(t, inner.Error(), err.Error())
```

Comment:
Could we add a sentence for this property in the `DeterministicError` docs? I think this is very powerful feature to have.